### PR TITLE
Add country summary endpoint and implement browse_rds

### PR DESCRIPTION
**Pull request by diversemix**
relates to BAK-1148

- Will supersede `document_browse` (which will need removing)
- Browse functionality not implemented (see BAK-1151)
- Country browsing implemented on the summary endpoint 
- Adds tests

### CODE REVIEW

**Comment by joel-wright** [comment]
We're going to have some fun merging this with what I've been up to... let's catch up in the morning (sorry :disappointed:)

File: backend/app/api/api_v1/routers/summaries.py [hunk 1]
```diff
@@ -25,8 +26,25 @@ def search_by_country(
...
+
+    for cat in CategoryName:
+        results = browse_rds(db, BrowseArgs(geography_id=geography_id, category=cat))
+        document_counts[cat] = len(results.documents)
+        top_documents[cat] = list(results.documents[:5])
+
+    events = get_events_for_country(db, geography_id)
+
+    # TODO: Add targets
+    targets = []
```
Comment (joel-wright) [inline backend/app/api/api_v1/routers/summaries.py:43]: ugh, targets... something else we need to make a plan for asap!

File: backend/app/core/browse.py [hunk 2]
```diff
@@ -1,12 +1,104 @@
...
+        document_country_english_shortname=row["country_name"],
+        # Stuff we don't use
+        document_source_name="",
+        document_type="",
+        document_source_url="",
+        document_url="",
+        document_content_type="",
+        document_title_match=False,
+        document_description_match=False,
+        document_passage_matches=[],
```
Comment (joel-wright) [inline backend/app/core/browse.py:43]: this looks so much like a `DocumentOverviewResponse`... perhaps we should be trying to unify them...?

**Review by joel-wright — APPROVED** [review]
LGTM, just a couple of questions

File: backend/app/core/browse.py (hunk 2 above)
Comment (diversemix) [reply backend/app/core/browse.py:43]: Let's chat - as this is for a response the same as search (as browse needs to return the same structure as search I think) .... /cc @PatFawbertMills

File: backend/app/api/api_v1/routers/summaries.py (hunk 1 above)
Comment (diversemix) [reply backend/app/api/api_v1/routers/summaries.py:43]: lol ... I did mention that to Michal too.

File: backend/app/core/browse.py (hunk 2 above)
Comment (diversemix) [reply backend/app/core/browse.py:43]: Talked - this will be coming with BAK-1151

**Pull request merged**